.github/workflows/ci_tests.yml

```diff
@@ -46,59 +49,92 @@ jobs:
-          - platform: ubuntu-latest
-            compiler:
+          - platform:
+              description: "ubuntu gcc"
               cpp: g++
               c: gcc
+              os: ubuntu-latest
             cpp_version: 17
             cmake_args:
               description: "Dynamic"
               args: "-DBUILD_SHARED_LIBS=on"
+            # MSVC has limited support for sanitizers, so we use include here for MSVC as an exception.
+            # This should be simplified after beman-project/exemplar#76 is merged.
+          - platform:
+              description: "windows MSVC"
+              cpp: cl
+              c: cl
+              os: windows-latest
+            cpp_version: 17
+            cmake_args:
+              description: "Default"
+              args: ""
+          - platform:
+              description: "windows MSVC"
+              cpp: cl
+              c: cl
+              os: windows-latest
+            cpp_version: 17
+            cmake_args:
+              description: "ASan"
+              # Debug infomation needed to avoid cl: C5072
+              # https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-c5072?view=msvc-170
+              args: "-DCMAKE_CXX_FLAGS='/fsanitize=address /Zi'"
+
 
-    name: "Unit: ${{ matrix.compiler.c }} ${{ matrix.cpp_version }} ${{ matrix.cmake_args.description }}"
-    runs-on: ${{ matrix.platform }}
+    name: "Unit: ${{ matrix.platform.description }} ${{ matrix.cpp_version }} ${{ matrix.cmake_args.description }}"
+    runs-on: ${{ matrix.platform.os }}
     steps:
       - uses: actions/checkout@v4
       - name: Install Ninja
         uses: lukka/get-cmake@latest
         with:
           cmakeVersion: "~3.25.0"
           ninjaVersion: "^1.11.1"
+      - name: Setup MSVC
+        if: startsWith(matrix.platform.os, 'windows')
+        uses: TheMrMilchmann/setup-msvc-dev@v3
+        with:
+          arch: x64
       - name: Print installed softwares
         run: |
-          clang++ --version
-          g++ --version
+          echo "Compiler:"
+          ${{ matrix.platform.cpp }} --version
+          ${{ matrix.platform.c }} --version
+
+          echo "Build system:"
           cmake --version
           ninja --version
       - name: Configure CMake
         run: |
           cmake -B build -S . -DCMAKE_CXX_STANDARD=${{ matrix.cpp_version }} ${{ matrix.cmake_args.args }}
         env:
-          CC: ${{ matrix.compiler.c }}
-          CXX: ${{ matrix.compiler.cpp }}
+          CC: ${{ matrix.platform.c }}
+          CXX: ${{ matrix.platform.cpp }}
           CMAKE_GENERATOR: "Ninja Multi-Config"
       - name: Build Release
         run: |
           cmake --build build --config Release --verbose
           cmake --build build --config Release --target all_verify_interface_header_sets
           cmake --install build --config Release --prefix /opt/beman.exemplar
-          find /opt/beman.exemplar -type f
+          tree /opt/beman.exemplar
```

Comment:
Maybe something like this?
```suggestion
          cd /opt/beman.exemplar
          tree
          cd -
```

Reply:
that's a great idea!

Reply:
Switched to a GitHub action runner provided directory. Still got empty directories from tree command, I don't know if `tree` is just badly implemented under bash environment

Reply:
Agh, it's a quirk with powershell tree.

See: https://superuser.com/questions/1270040/windows-10-powershell-tree-command-how-to-show-files

Reply:
switch out to `ls -R`, and it is working. It's just very ugly...

![image](https://github.com/user-attachments/assets/92e1ed7e-19d8-4e57-855e-1974d47ef984)
![image](https://github.com/user-attachments/assets/5461e89a-658d-44db-9ad6-40f07fb10908)

Reply:
You can use the same trick I suggested in [this comment](https://github.com/bemanproject/exemplar/pull/80#discussion_r1870168036) to use the `/F` flag on windows only